### PR TITLE
Kwabena/add png support

### PR DESCRIPTION
**Pull request by bitbank2**
This set of changes enables initial PNG decode support. Most pixel source/dest types are supported.

### CODE REVIEW

**Review by kwagyeman — APPROVED** [review]

File: src/omv/imlib/zbar.c [hunk 1]
```diff
@@ -8714,18 +8714,7 @@ void zbar_scanner_get_state (const zbar_scanner_t *scn,
 void imlib_find_barcodes(list_t *out, image_t *ptr, rectangle_t *roi)
 {
     uint8_t *grayscale_image = (ptr->pixfmt == PIXFORMAT_GRAYSCALE) ? ptr->data : fb_alloc(roi->w * roi->h, FB_ALLOC_NO_HINT);
```
Comment (kwagyeman) [inline src/omv/imlib/zbar.c:8716]: Undo this change.

File: src/omv/imlib/png.c [hunk 2]
```diff
@@ -52,18 +117,196 @@ bool png_compress(image_t *src, image_t *dst, bool realloc)
...
+    return; // good decode
 }
 
 #if defined(IMLIB_ENABLE_IMAGE_FILE_IO)
 // This function inits the geometry values of an image.
 void png_read_geometry(FIL *fp, image_t *img, const char *path, png_read_settings_t *rs)
 {
-    // Larry, see the jpeg version for how to populate the png read settings
-}
+    for (;;) {
```
Comment (kwagyeman) [inline src/omv/imlib/png.c]: Why is the for(;;) needed.

File: src/omv/imlib/zbar.c (hunk 1 above)
Comment (bitbank2) [reply src/omv/imlib/zbar.c:8716]: How do I undo a non-change change?

File: src/omv/imlib/png.c (hunk 2 above)
Comment (bitbank2) [reply src/omv/imlib/png.c]: It was a carry over from the JPEG version; fixed now.

**Review by kwagyeman — APPROVED** [review]
